test/models/binomial.py

```diff
@@ -0,0 +1,72 @@
+from typing import Union
+from scipy import stats
+from scipy.special import logit, expit as inv_logit, log1p
+import numpy as np
+import numpy.typing as npt
+
+
+class Binomial:
+    """
+    Binomial model with a (conjugate) beta prior.
+
+    This model is decomposed as a prior and a likelihood for
+    testing with samplers that require that, e.g. SMC. The
+    posterior has a closed-form beta distribution.
```

Comment:
We only hyphenate noun compounds in English and then only when used as adjectives (unless there's a standard hyphenated form).  The word "closed" is an adjective here.  Don't blame me, I didn't make up these whacky rules!  Online you see it both ways, so no big deal for this one, just wanted to throw out the general rule (which I often have a hard time following myself).